### ISSUE TITLE
New Role: Test Case Creator

### ISSUE THREAD

**Issue by brunosmm** [gsmainclusivetechlab/interop-test-platform#135]
Will be able to perform the same as a normal user and upload a test case to draft mode, once the test case is uploaded automatic assign it to this user.

Access to test case pages, but not able to remove, only create.
https://interop.gsmainclusivetechlab.io/admin/scenarios/1/test-cases

**Comment by jcakokorev** [comment]
@brunosmm Test Case Creator can:

- View this page http://staging.interop.gsmainclusivetechlab.io/admin/test-cases with all (public/private) or only own test cases?
- Ability to assign all (public/private) or only own test cases to own session?
- Ability to import test case here http://staging.interop.gsmainclusivetechlab.io/admin/test-cases/import
- Don't have permissions to remove or update test case

P.S. First and second options i think depends each other.

**Comment by cjol** [comment]
I just talked to Bruno about test case creator permissions, here is what we have decided. Test case creator can:

 - view test case admin page
 - View all test cases that a normal user would see (i.e. public test cases, and assigned test cases once https://github.com/gsmainclusivetechlab/interop-test-platform/issues/134 is implemented)
 - Also view private test cases for which they are the owner
 - Create sessions using test cases like a normal user (i.e. public and assigned test cases as above)
 - Also create sessions using private test cases for which they are the owner
 - Create (import) new test cases with themselves as owner
 - Cannot update or delete any public test cases
 - Can update or delete private test cases for which they are the owner

That's a slight difference from what was initially outlined, but hopefully that's clear.
